c/tskit/trees.c

```diff
@@ -2103,22 +2126,635 @@ tsk_treeseq_sample_count_stat(const tsk_treeseq_t *self, tsk_size_t num_sample_s
         ret = TSK_ERR_NO_MEMORY;
         goto out;
     }
-    j = 0;
-    for (k = 0; k < num_sample_sets; k++) {
-        for (l = 0; l < sample_set_sizes[k]; l++) {
-            u = sample_sets[j];
-            sample_index = self->sample_index_map[u];
-            weight_row = GET_2D_ROW(weights, num_sample_sets, sample_index);
+    ret = init_sample_weights(
+        self, num_sample_sets, sample_set_sizes, sample_sets, &weights);
+    if (ret != 0) {
+        goto out;
+    }
+    ret = tsk_treeseq_general_stat(self, num_sample_sets, weights, result_dim, f, &args,
+        num_windows, windows, options, result);
+out:
+    tsk_safe_free(weights);
+    return ret;
+}
+
+/***********************************
+ * Two Locus Statistics
+ ***********************************/
+
```

Comment:
Note that the pass by ref/value is a bit different from `get_allele_weights`. My version does not have any allocation and modifies the passed array directly. I couldn't quite figure out why `get_allele_weights` was designed the way it was.

Reply:
I'm not sure either - let's discuss tomorrow!